File: src/app/topics/mock-topics.ts
import {TopicModel} from '../model/topic.model';
import {QuestionModel} from '../model/question.model';
import {AnswerModel} from '../model/answer.model';

const a1: Array<AnswerModel> = [
  new AnswerModel('red'),
  new AnswerModel('green'),
  new AnswerModel('blue')
];
const a2: Array<AnswerModel> = [
  new AnswerModel('cappuccino'),
  new AnswerModel('latte'),
  new AnswerModel('espresso')
];
const q1: Array<QuestionModel> = [
  new QuestionModel('Your favourite colour?', 'single-choice', a1)
];
const q2: Array<QuestionModel> = [
  new QuestionModel('Your favourite coffee?', 'single-choice', a2)
];
export const TOPICS: TopicModel[] = [
  new TopicModel(0,'Colors', q1, true),
  new TopicModel(1,'Coffee', q2, true)
];

File: fapi/src/main/java/fapi/service/mock/MockTopics.java
package fapi.service.mock;



import fapi.models.AnswerModel;
import fapi.models.QuestionModel;
import fapi.models.TopicModel;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;


public class MockTopics {
  private static List<TopicModel> mockTopics = new ArrayList<>(Arrays.asList(
    new TopicModel(0, "Colors", new ArrayList<>(Arrays.asList(
      new QuestionModel("Your favourite colour?", "single-choice", new ArrayList<>(Arrays.asList(
        new AnswerModel("red"), new AnswerModel("green"), new AnswerModel("blue")
      ))),
      new QuestionModel("Color of your eyes?", "single-choice", new ArrayList<>(Arrays.asList(
        new AnswerModel("green"), new AnswerModel("hazel"), new AnswerModel("light blue")
      )))
    )), true),
    new TopicModel(1, "Tea", new ArrayList<>(Arrays.asList(
      new QuestionModel("Do you drink tea?", "single-choice", new ArrayList<>(Arrays.asList(
        new AnswerModel("yes"), new AnswerModel("no")
      ))),
      new QuestionModel("Which tea do you prefer?", "multiple-choice", new ArrayList<>(Arrays.asList(
        new AnswerModel("green"), new AnswerModel("black"), new AnswerModel("red"), new AnswerModel("white")
      )))
    )), true)

  ));

  public static void addTopic(TopicModel t) {
    mockTopics.add(t);
  }
  public static void deleteTopic(Long id) {mockTopics.removeIf(t -> t.getId() == id);}
  public static TopicModel getTopic(Long id) {
    return mockTopics.stream().filter(t -> t.getId() == id).findFirst().get();
  }
  public static List<TopicModel> getMockTopics() {
    return mockTopics;
  }
}
